libs/game/sim/multiplayer.ts

```diff
@@ -0,0 +1,134 @@
+namespace pxsim.multiplayer {
+    export function postImage(im: pxsim.RefImage, goal: string) {
+        const asBuf = pxsim.image.toBuffer(im);
+        getMultiplayerState().send(<MultiplayerImageMessage>{
+            content: "Image",
+            image: asBuf,
+            goal
+        });
+    }
+
+    export function getCurrentImage(): pxsim.RefImage {
+        return getMultiplayerState().backgroundImage;
+    }
+
+    export function setOrigin(origin: "client" | "server" | undefined) {
+        getMultiplayerState().origin = origin;
+    }
+
+    export function getOrigin(): string {
+        return getMultiplayerState().origin;
+    }
+}
+
+namespace pxsim {
+    export interface MultiplayerBoard extends EventBusBoard {
+        multiplayerState: MultiplayerState;
+    }
+
+    export function getMultiplayerState() {
+        return (board() as EventBusBoard as MultiplayerBoard).multiplayerState;
+    }
+
+    export interface SimulatorMultiplayerMessage extends SimulatorBroadcastMessage {
+        broadcast: true
+        type: "multiplayer";
+        content: string;
+        origin?: "server" | "client";
+        clientNumber?: number;
+        id?: number;
+    }
+
+    export interface MultiplayerImageMessage extends SimulatorMultiplayerMessage {
+        content: "Image";
+        goal: string; // goal of message; e.g. "broadcast-screen"
+        image: RefImage;
+    }
+
+    export interface MultiplayerButtonEvent extends SimulatorMultiplayerMessage {
+        content: "Button";
+        button: number; // pxsim.Key.A, ...
+        state: "Pressed" | "Released" | "Held";
+    }
+
+    type MultiplayerOrigin = "client" | "server" | undefined;
+
+    export class MultiplayerState {
+        lastMessageId: number;
+        origin: "client" | "server" | undefined;
```

Comment:
I made a slight change here so the sim will start up with origin set to undefined (i.e. not in a multiplayer mode). This ensures no multiplayer messages are sent.